### PR TITLE
LIVY-208. Exposing YARN driver log url and Spark UI url for YARN sessions.

### PR DESCRIPTION
**Pull request by tc0312**
They are exposed as:
- appInfo.driverLogUrl
- appInfo.sparkUiUrl

### CODE REVIEW

**Comment by tc0312** [comment]
@vanzin @jerryshao @zjffdu Mind if you guys take a look?

File: server/src/main/scala/com/cloudera/livy/utils/SparkApp.scala [hunk 1]
```diff
@@ -18,15 +18,25 @@
...
 package com.cloudera.livy.utils
 
+import scala.collection.JavaConverters._
+
 import com.cloudera.livy.LivyConf
 import com.cloudera.livy.util.LineBufferedProcess
 
+case class AppInfo(var driverLogUrl: Option[String] = None, var sparkUiUrl: Option[String] = None) {
+  def asJavaMap: java.util.Map[String, String]
+    = Map("driverLogUrl" -> driverLogUrl.orNull, "sparkUiUrl" -> sparkUiUrl.orNull).asJava
```
Comment (jerryshao) [inline server/src/main/scala/com/cloudera/livy/utils/SparkApp.scala]: nit: move `=` to the above line.

Comment (jerryshao) [inline server/src/main/scala/com/cloudera/livy/utils/SparkApp.scala]: Can we define these keys "driverLogUrl", "sparkUiUrl" in a common place. I found that lots of places use them, and it easily introduces the typo.

File: server/src/main/scala/com/cloudera/livy/utils/SparkYarnApp.scala [hunk 2]
```diff
@@ -209,16 +211,43 @@ class SparkYarnApp private[utils] (
...
+          changeState(mapYarnState(
+            appReport.getApplicationId,
+            appReport.getYarnApplicationState,
+            appReport.getFinalApplicationStatus))
+
+          val latestAppInfo = {
+            val attempt =
+              yarnClient.getApplicationAttemptReport(appReport.getCurrentApplicationAttemptId)
+            val driverLogUrl =
+              Try(yarnClient.getContainerReport(attempt.getAMContainerId).getLogUrl)
```
Comment (jerryshao) [inline server/src/main/scala/com/cloudera/livy/utils/SparkYarnApp.scala]: Does it mean driver should be run on AM, here the driver log is actually AM log? So in that case, how can we handle yarn client mode, or other cluster manager like Standalone or Mesos?

File: integration-test/src/test/scala/com/cloudera/livy/test/BatchIT.scala [hunk 3]
```diff
@@ -56,6 +56,10 @@ class BatchIT extends BaseIntegrationTestSuite with BeforeAndAfterAll {
     dumpLogOnFailure(result.id) {
       assert(result.state === SessionState.Success().toString())
       assert(cluster.fs.isDirectory(new Path(output)))
+      result.appInfo should contain key ("driverLogUrl")
+      result.appInfo should contain key ("sparkUiUrl")
+      result.appInfo.get("driverLogUrl")
```
Comment (zjffdu) [inline integration-test/src/test/scala/com/cloudera/livy/test/BatchIT.scala]: duplicated test with `result.appInfo should contain key ("driverLogUrl")`

File: server/src/main/scala/com/cloudera/livy/utils/SparkYarnApp.scala (hunk 2 above)
Comment (tc0312) [inline server/src/main/scala/com/cloudera/livy/utils/SparkYarnApp.scala]: You are right and Livy officially supports yarn-cluster mode and local mode only.

**Comment by tc0312** [comment]
Fixed. Please review.

**Comment by jerryshao** [comment]
LGTM.

**Pull request merged**